### DISCUSSION TITLE
Upgrading to `core` 0.16 and removing `core_kernel`?

### DISCUSSION THREAD

**Discussion by L-as** [MinaProtocol/mina#13425] [General]
I noticed Mina is on 0.14.1, and am wondering how hard it would be to upgrade to `core` 0.16. I am not experienced with OCaml, hence asking to roughly estimate the size of the work.

**Comment by psteckler** [comment]
It may or may not be possible, depending on the dependencies of other packages. Someone would have to try to upgrade locally, see what OPAM reports.

You'd need a new toolchain image, maybe others.

**Comment by psteckler** [comment]
Also, some repo code would likely need to change due to API changes.
